pkg/backup/restore_planning.go

```diff
@@ -1686,6 +1688,33 @@ func doRestorePlan(
 
 	var fullyResolvedSubdir string
 
+	if restoreStmt.Options.OnlineImpl() {
+		// validate uris are allowed for online restore
+		for _, path := range from {
+			uri, err := url.Parse(path)
+			if err != nil {
+				return err
```

Comment:
nit: call errors.Wrap on this error. This is especially important for errors that come from the stdlib because we need to wrap the error to get a useful stack trace.